doc/terminology.rst

```diff
@@ -0,0 +1,51 @@
+.. _terminology:
+
+.. https://github.com/pydata/xarray/issues/2410
+.. https://github.com/pydata/xarray/issues/1295
+
+Terminology
+===========
+
+*Xarray terminology differs slightly from CF and mathematical conventions, and therefore using xarray, understanding the documentation, and parsing error messages is easier once key terminology is defined. This glossary was designed so that more fundamental concepts come first. Thus for new users, this page is best read top-to-bottom. Throughout the glossary,* ``arr`` *will refer to an xarray* :py:class:`DataArray` *in any small examples. For more complete examples, please consult the relevant documentation.*
+
+----
+
+**DataArray:** A multi-dimensional array with labeled or named dimensions. If its optional ``name`` property is set, it is a *named DataArray*.
+
+----
+
+**Dimension / dimensions:** A *dimension* is a nonnegative number for the dimensionality of the underlying data, while an array's *dimensions* are a set of dimension names. The name of the ``i``-th dimension is ``arr.dims[i]``. If an array is created without dimensions, the default dimension names are ``dim_0``, ``dim_1``, and so forth.
+
+----
+
+**Coordinate:** A one-dimensional array that labels a dimension of another ``DataArray``. There are two types of coordinate arrays: *dimension coordinates* and *non-dimension coordinates* (see below). A coordinate named ``x`` can be retrieved from ``arr.coords[x]``. A ``DataArray`` can have more coordinates than dimensions because a single dimension can be assigned multiple coordinate arrays. However, only one coordinate array can be a assigned as a particular dimension's dimension coordinate    array. As a consequence, ``len(arr.dims) <= len(arr.coords)`` in general.
```

Comment:
non-dimensional coordinates can be multi-dimensional

Reply:
I thought "multi-dimensional coordinates" really just meant multiple one-dimensional coordinate arrays labeling the same dimension. I inferred that from [this page](https://xarray.pydata.org/en/stable/examples/multidimensional-coords.html). Is this what you mean or something else? Also, could you please show me an example? I've tried and failed to use `assign_coords` to add a multi-dimensional `DataArray` as a coordinate array.

Reply:
For example

```python

In [8]: xr.DataArray(np.ones((2,3)), dims=['a','b'], coords=dict(c=(('a','b'), np.zeros((2,3)))))
Out[8]:
<xarray.DataArray (a: 2, b: 3)>
array([[1., 1., 1.],
       [1., 1., 1.]])
Coordinates:
    c        (a, b) float64 0.0 0.0 0.0 0.0 0.0 0.0
Dimensions without coordinates: a, b
```